src/ui/Facet/FacetSearch.ts

```diff
@@ -325,6 +325,15 @@ export class FacetSearch implements IFacetSearch {
     });
   }
 
+  private appendSelectAllResultsButton() {
+    const selectAll = document.createElement('li');
+    $$(selectAll).addClass(['coveo-facet-selectable', 'coveo-facet-search-selectable', 'coveo-facet-search-select-all']);
+    $$(selectAll).text(l('SelectAll'));
+    $$(selectAll).setAttribute('aria-hidden', 'true');
+    $$(selectAll).on('click', () => this.selectAllValuesMatchingSearch());
```

Comment:
Should this be selectable using something else than click ?

Would using the `AccessibleButton` class/utility be a good idea here ?

Reply:
In addition to `click`, the button can be selected using the `Enter` key. However, the event handler is not on the element. Rather it is on the facet search element which maintains focus when navigating between the facet search results using a keyboard.

I compared against other sites, and keeping the focus on the search element looks like the best-practice and the right design choice.

I do not think using an `AccessibleButton` is needed here. In fact I added the `aria-hidden` attribute on line 332 to prevent the screen reader from reading the text twice. The read events are managed by the dropdown navigator class.